src/cert_provider/CMakeLists.txt

```diff
@@ -11,8 +11,21 @@ add_dependencies(build_tests aktualizr-cert-provider)
 
 install(TARGETS aktualizr-cert-provider RUNTIME DESTINATION bin)
 
-add_test(NAME certprovider-ca COMMAND ${PROJECT_SOURCE_DIR}/tests/run_certprovider_test.sh
-        ${PROJECT_BINARY_DIR}/src/cert_provider/aktualizr-cert-provider ${PROJECT_SOURCE_DIR}/tests/test_data/credentials.zip
-        ${PROJECT_SOURCE_DIR}/tests/test_data/CAcert.pem ${PROJECT_SOURCE_DIR}/tests/test_data/CApkey.pem)
+set(CREDS ${SOTA_PACKED_CREDENTIALS})
+
+if (TESTSUITE_EXCLUDE)
+  foreach(label ${TESTSUITE_EXCLUDE})
+    if (${label} STREQUAL "credentials")
+      set(CREDS "")
+    endif()
+  endforeach()
+endif(TESTSUITE_EXCLUDE)
```

Comment:
I think this can be done a lot more easily by setting the right LABEL for this test target, for example: `set_tests_properties(test_aktualizr_cert_provider PROPERTIES LABELS "credentials")`. You can see an example of this in `src/libaktualizr/uptane/CMakeLists.txt`. If you do that, then the Travis script that runs the tests will automatically exclude it because of the `-eTEST_TESTSUITE_EXCLUDE=credentials` parameter given to cmake.

Reply:
Yes, but it will require splitting the tests into two files, since the aforementioned mechanism works per test/file bases. I'll split the tests into two parts as we discussed offline and apply the described approach.

Reply:
Actually, splitting tests into two files look more complicated compared to what is implemented in this CMakeLists.

Reply:
Why? I expect that some code will need to be put in a common location (possibly the already available TestUtils), but what else has to be taken care of?

Reply:
Also not a big fan but it should at least have an explanatory comment in the CMake definition.

Other points:

- use https://cmake.org/cmake/help/latest/command/list.html#search maybe?
- the skipped test silently succeeding with empty credentials is very misleading

Reply:
The list search doesn't work for a variable that contains just a single value for some reason.

Yeah, anyway, I agree that it's better actually split these tests into two parts, it just look more readable and better/easier extendable compared to what is now.

Reply:
Moreover, just setting the label to a test does not help to exclude the test. TESTSUITE_EXCLUDE definition is used/applied solely to tests directory and corresponding ctest parameters are applied only to 'check' target.

Reply:
What do you mean "it does not help to exclude the test"?

It's directly supported by the `check` target, as you've said, which is the recommended way to run the suite (and what is used by CI) and it also works for tests defined outside of the `tests` directory, as you can see on CI.

The reason we have this surprising setup is that we had the problem of tests requiring credentials only being compiled when `SOTA_PACKED_CREDENTIALS` was defined. As we used to not have credentials on our only CI at the time (Travis), and as developers tended to not run the testsuite with the variable either, these tests often weren't compiled at all and broke often. Also in theory, they'd need to be compiled to be checked by clang-tidy (but we disabled the checks for the unit tests IIRC).

So, the solution was to define bogus credentials in Travis, to compile the test but not running them.